rust/lance/src/utils/future.rs

```diff
@@ -45,7 +48,10 @@ impl<T: Clone> SharedPrerequisite<T> {
             .get()
             .await
             .map(|_| ())
-            .map_err(|err| crate::Error::PrerequisiteFailed { message: err })
+            .map_err(|err| crate::Error::PrerequisiteFailed {
+                message: err,
+                location: location!(),
```

Comment:
Can we reuse the inner error's location here?

Reply:
It is hard to reuse location here because it is hard to add `Clone` for `Error`